configargparse.py

```diff
@@ -813,19 +927,68 @@ def parse_known_args(
             env_vars=os.environ,
             ignore_help_args=False,
     ):
-        """Supports all the same args as the `argparse.ArgumentParser.parse_args()`,
+        """Supports all the same args as the `argparse.ArgumentParser.parse_known_args()`,
         as well as the following additional args.
 
         Arguments:
             args: a list of args as in argparse, or a string (eg. "-x -y bla")
             config_file_contents (str). Used for testing.
             env_vars (dict). Used for testing.
-            ignore_help_args (bool): This flag determines behavior when user specifies ``--help`` or ``-h``. If False,
-                it will have the default behavior - printing help and exiting. If True, it won't do either.
+            ignore_help_args (bool): This flag determines behavior when user specifies ``--help``
+                or ``-h``. If False, it will have the default behavior - printing help and exiting.
+                If True, it won't do either.
 
         Returns:
             tuple[argparse.Namespace, list[str]]: tuple namescpace, unknown_args
         """
+        return self._parse_known_args_cap(intermixed=False,
+                                          args=args,
+                                          namespace=namespace,
+                                          config_file_contents=config_file_contents,
+                                          env_vars=env_vars,
+                                          ignore_help_args=ignore_help_args)
+
+    def parse_known_intermixed_args(
```

Comment:
sorry, naive question until I have more time to dig into the differences - why are two separate sets of methods needed for intermixed and non-intermixed args? Why can't all cases be handled as intermixed?

Reply:
A very good question, and TBH I'm not entirely sure of the full details. I followed the underlying `argparse.py` code and just made sure the upstream tests were working if I call the corresponding function in the superclass. If I force everything to be intermixed then I get a *lot* of test failures (from `python3 -munittest tests.test_argparse_tests`). The [upstream docs](https://docs.python.org/3/library/argparse.html#intermixed-parsing) say, amongst other things:

> These \[intermixed\] parsers do not support all the argparse features, and will raise exceptions if unsupported 
> features are used. In particular, subparsers, and mutually exclusive groups that include both optionals and 
> positionals are not supported.